scripts/test_setup.py

```diff
@@ -1,23 +1,62 @@
 import argparse
 import configparser
 import contextlib
+import json
 import os
 import requests
 import subprocess
+from pathlib import Path
 
 parser = argparse.ArgumentParser()
+parser.add_argument('env')
 parser.add_argument('--tag', action='store_true')
 
 
-def version_info(channels):
-    conda_list = subprocess.check_output(['conda', 'list'],
+def version_info():
+    conda_list = subprocess.check_output(['conda', 'list', '--json'],
                                          universal_newlines=True)
-    info = {}
-    for line in conda_list.split('\n'):
-        if any(ch in line for ch in channels):
-            pkg, ver, _, _ = line.split()
-            info[pkg] = ver
-    return info
+    info_list = json.loads(conda_list)
+    version_dict = {}
+
+    for item in info_list:
+        version_dict[item['name']] = item['version']
+
+    return version_dict
+
+
+def setup_tests(repo_file, tags=None):
+    url_base = 'https://github.com/{}.git'
+    repo_file = Path(repo_file)
+
+    with repo_file.open('r') as fd:
+        repos = fd.read().strip().split('\n')
```

Comment:
Not necessary to change, but a convenience method I've been moving to recently:
```suggestion
        repos = fd.read().strip().splitlines()
```

Reply:
That's a nifty one